### PR TITLE
Bump powsybl-core 4.3.0-alpha-1

### PR DESCRIPTION
**Pull request by flo-dup**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** 
No

**What kind of change does this PR introduce?**
Update to new API

**What is the current behavior?**
Old security analysis API is used

**What is the new behavior (if this is a feature change)?**
New security analysis API is used

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:
See powsybl/powsybl-core#1716

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisProvider.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    final Supplier<GraphDecrementalConnectivity<LfBus>> connectivityProvider;
+
+    public OpenSecurityAnalysisProvider(MatrixFactory matrixFactory, Supplier<GraphDecrementalConnectivity<LfBus>> connectivityProvider) {
+        this.matrixFactory = matrixFactory;
+        this.connectivityProvider = connectivityProvider;
+    }
+
+    public OpenSecurityAnalysisProvider() {
+        this(new DenseMatrixFactory(), () -> new NaiveGraphDecrementalConnectivity<>(LfBus::getNum));
```
Comment (geofjamg) [inline src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisProvider.java:44]: In the actual code it is EvenShiloachGraphDecrementalConnectivity

File: src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisProvider.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+@AutoService(SecurityAnalysisProvider.class)
+public class OpenSecurityAnalysisProvider implements SecurityAnalysisProvider {
+
+    final MatrixFactory matrixFactory;
```
Comment (geofjamg) [inline src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisProvider.java:34]: private

File: src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisProvider.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    final Supplier<GraphDecrementalConnectivity<LfBus>> connectivityProvider;
```
Comment (geofjamg) [inline src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisProvider.java:36]: private

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisProvider.java (hunk 1 above)
Comment (flo-dup) [reply src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysisProvider.java:44]: Corrected here and in run.

**Review by annetill — APPROVED** [review]

**Pull request merged**